ts/components/SetupWidget.tsx

```diff
@@ -97,6 +106,18 @@ export default class SetupWidget extends React.Component<SetupWidgetProps, Setup
 		this.loadFile(file);
 	}
 
+	protected preloadReplay(file: string) {
+		this.setState({ working: true });
+		fs.readFile(file, "utf8", (err, data) => {
+  			if (err) {
+  				this.setState({ working: false })
+  				console.error(err)
```

Comment:
there don't seem to be any error handling in `SetupWidget`, so i just followed the pattern

Reply:
Yeah, this is fine.